raiden/api/rest.py

```diff
@@ -94,88 +105,135 @@ class RestAPI(object):
 
         # wrap in EventsList:
-        events_list = EventsList(api_result)
+        events_list = EventsList(raiden_service_result)
         result = self.events_list_schema.dumps(events_list)
         return result
 
-    def patch_channel(self, channel_address, deposit=None, status=None):
-
-        if deposit is not None and status is None:
-            api_result = self.raiden_api.deposit(deposit)
-            return self.channel_schema.dumps(api_result)
-
-        elif status is not None and deposit is None:
-
-            if status == 'closed':
-                api_result = self.raiden_api.close(channel_address)
-                return self.channel_schema.dumps(api_result)
+    def get_channel(self, channel_address):
+        channel = self.raiden_api.get_channel(channel_address)
+        return self.channel_schema.dumps(channel)
+
+    def get_partners_by_token(self, token_address):
+        return_list = []
+        raiden_service_result = self.raiden_api.get_channel_list(token_address)
+        for result in raiden_service_result:
+            return_list.append({
+                'partner_address': result.partner_address,
+                'channel': '/api/{}/channels/{}'.format(
+                    self.version,
+                    address_encoder(result.channel_address)
+                )
+            })
+        # wrap in the list schema
+        schema_list = PartnersPerTokenList(return_list)
+        result = self.partner_per_token_list_schema.dumps(schema_list)
+        return result
 
-            if status == 'settled':
-                api_result = self.raiden_api.close(channel_address)
-                return self.channel_schema.dumps(api_result)
+    def patch_channel(self, channel_address, balance=None, state=None):
+        # find the channel
+        channel = self.raiden_api.get_channel(channel_address)
 
-            if status == 'open':
+        # if we patch with `balance` it's a deposit
+        if balance is not None and state is None:
+            raiden_service_result = self.raiden_api.deposit(
+                channel.token_address,
+                channel.partner_address,
+                balance
+            )
+            return self.channel_schema.dumps(raiden_service_result)
+
+        elif state is not None and balance is None:
+
+            if state == 'closed':
+                raiden_service_result = self.raiden_api.close(
+                    channel.token_address,
+                    channel.partner_address
+                )
+                return self.channel_schema.dumps(raiden_service_result)
+            elif state == 'settled':
+                raiden_service_result = self.raiden_api.settle(
+                    channel.token_address,
+                    channel.partner_address
+                )
+                return self.channel_schema.dumps(raiden_service_result)
+            elif state == 'open':
                 raise Exception('nothing to do here')
+
         raise Exception()
```

Comment:
Please, use `abort()` instead of raising a plain exception and add a `message` informing the client that it doesn't make sense to increase the balance and then close the channel. eg. `abort(httplib.CONFLICT, message='cannot update balance and close at same time.')`

[409 CONFLICT](https://httpstatuses.com/409)